### PR TITLE
docs: integrate package documentation

### PR DESCRIPTION
**Pull request by Vincinator**
**What this PR does**:

To integrate the new package documentation (also included in this PR), I took the freedom to rearrange the structure of the docs. 

- Added docs on how to create a package from git source
- Added docs on how to create a new package version that is compatible with old runtime dependencies (e.g. glibc)
- Consolidated FAQ topics into FAQ
- Created docs/build/troubleshooting.md for issues related to user build environments 
- Re-worked docs/build/README.md  
    - To avoid outdated documentation we should not re-specify available build.sh flags here. 
    
Related: #740  

**Notes for Reviewer**
I find it difficult to read my changes of this PR via the diff UI of "Files Changed"..
Files to read:
- [docs/FAQ.md](https://github.com/gardenlinux/gardenlinux/blob/docs/repo-packages/docs/FAQ.md)
- [docs/build/README.md](https://github.com/gardenlinux/gardenlinux/blob/docs/repo-packages/docs/build/README.md)
- [docs/build/troubleshooting.md](https://github.com/gardenlinux/gardenlinux/blob/docs/repo--packages/docs/build/troubleshooting.md)
- [docs/repository/README.md](https://github.com/gardenlinux/gardenlinux/blob/docs/repo-packages/docs/repository/README.md)

### CODE REVIEW

File: docs/build/README.md [hunk 1]
```diff
@@ -1,89 +1,109 @@
...
+
+## Build Artifacts
+Build artifacts are stored in the output folder (default `.build/`).
+Some artifacts will only be created by certain features. 
+
+# Kernel Module
+Drivers/LKMs not included in upstream linux of kernel.org can be build out of tree.
+
+## build-kernelmodule container 
+We provide build container that come with gardenlinux linux-headers installed.
```
Comment (MrBatschner) [inline docs/build/README.md:44]:
```suggestion
We provide a build container that come with Garden Linux linux-headers installed. 
```

File: docs/build/README.md [hunk 2]
```diff
@@ -1,89 +1,109 @@
...
+These build containers have a `uname -r` wrapper installed. 
+This wrapper outputs the latest installed kernel header in that container.
+
+Container is created here: https://gitlab.com/gardenlinux/driver/gardenlinux-driver-build-container
+
+```
+docker pull registry.gitlab.com/gardenlinux/driver/gardenlinux-driver-build-container/gl-driver-build:dev
+```
+
+1. Load your kernel module sources into the container
```
Comment (MrBatschner) [inline docs/build/README.md:54]: In .md files, for a numbered list, prefix all the bullet points with just `1.` and leave the numbering to the markdown renderer.

**Review by MrBatschner — CHANGES_REQUESTED** [review]
First batch of review comments, more to come. Some typos and corrections. 
In general, run a `s/debian/Debian/g` and `s/garden\s*linux/Garden Linux/gI` on the text. ;-)

File: docs/build/README.md [hunk 3]
```diff
@@ -1,89 +1,109 @@
...
+	- [Build Artifacts](#build-artifacts)
+- [Kernel Module](#kernel-module)
+	- [build-kernelmodule container](#build-kernelmodule-container)
+- [Package Build](#package-build)
+	- [Git Source](#git-source)
+	- [Snapshot Source](#snapshot-source)
+
+# Introduction
+
+Garden Linux makes use of the Github Release feature. Checkout available Releases [here](https://github.com/gardenlinux/gardenlinux/releases).
```
[the remark this answers is not in the document]
Comment (Vincinator) [reply docs/build/README.md]: Agreed - applied feedback

**Review by MrBatschner — CHANGES_REQUESTED** [review]
Some more nitpicking.

**Review by MrBatschner — APPROVED** [review]
/lgtm

**Comment by Vincinator** [comment]
Solved merge conflict during rebase with main branch

**Pull request merged**